dateparser/date.py

```diff
@@ -120,11 +122,18 @@ def get_date_from_timestamp(date_string, settings, negative=False):
          match = RE_SEARCH_TIMESTAMP.search(date_string)
 
     if match:
+        if settings is not None and settings.TIMEZONE is not None and 'local' not in settings.TIMEZONE.lower():
```

Comment:
```suggestion
        if settings is not None and settings.TIMEZONE is not None and 'local' not in settings.TIMEZONE.lower():
```

I am unsure about `'local' not in settings.TIMEZONE.lower()`. According to the documentation, the local timezone is used by default (i.e. when the value is None), and otherwise a valid timezone has to be specified. So this statement does not seem necessary, and it can be troublesome (imagine a city containing the letters “local” in it).

Reply:
> I am unsure about 'local' not in settings.TIMEZONE.lower(). According to the documentation, the local timezone is used by default (i.e. when the value is None), and otherwise a valid timezone has to be specified. 

RIght, when `settings` is `None` (maybe this is impossible :shrug:), or the `TIMEZONE` is `None` or `'local'` is in the timezone, then it'll use `get_localzone()`. It's only when `settings.TIMEZONE` exists and does not contain `'local'`, that the code will call `get_timezone_from_tz_string(settings.TIMEZONE)`. Would it be better to flip the boolean logic and use:

```python
        if settings is None or settings.TIMEZONE is None or 'local' in settings.TIMEZONE.lower():
            local_timezone = get_localzone()
        else:
            local_timezone = get_timezone_from_tz_string(settings.TIMEZONE)
```

> So this statement does not seem necessary, and it can be troublesome (imagine a city containing the letters “local” in it).

I got that idiom from other parts of the code that check if the timezone is local:

```
dateparser/date_parser.py:        _settings_tz = settings.TIMEZONE.lower()
dateparser/date_parser.py:            if 'local' not in _settings_tz:
dateparser/date_parser.py:            if 'local' in _settings_tz:
dateparser/freshness_date_parser.py:        _settings_tz = settings.TIMEZONE.lower()
dateparser/freshness_date_parser.py:            if 'local' not in _settings_tz:
dateparser/freshness_date_parser.py:            if 'local' in _settings_tz:
dateparser/freshness_date_parser.py:            if 'local' not in _settings_tz:
dateparser/utils/__init__.py:    if 'local' in settings.TIMEZONE.lower():
```

Is there a better one to use?

Reply:
Hmm…

Well, if it is done elsewhere, I see no reason to block your pull request for that reason. Also, it does not seem like any [current timezone](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) contains the work “local” in it.